docs/reference-guides/best-practices/rancher-server/tuning-and-best-practices-for-rancher-at-scale.md

```diff
@@ -21,6 +21,26 @@ This guide describes the best practices and tuning approaches to scale Rancher s
 
 When scaling up Rancher, one typical bottleneck is resource growth in the upstream (local) Kubernetes cluster. The upstream cluster contains information for all downstream clusters. Many operations that apply to downstream clusters create new objects in the upstream cluster and require computation from handlers running in the upstream cluster.
 
+### Minimizing third-party software that runs on the Upstream Cluster
+
+Running Rancher at scale can put significant load on internal Kubernetes components such as `etcd` or `kubeapiserver`, therefore issues may arise if third party software interferes with the performance of those components or Rancher itself.
+
+In general, every third party piece of software carries a risk of interference, therefore to avoid performance issues it is recommended to limit software running on the Upstream Cluster to Kubernetes system components and Rancher itself as close as possible.
```

Comment:
```suggestion
Every third-party app carries a risk of interference. To prevent performance issues on the upstream cluster, you should avoid running any other apps, beyond Kubernetes system components and Rancher itself.
```

Reply:
I like the new wording, the only minor point I'd change is "software" instead of "app", as the former sounds more generic to me, encompassing things like components which are not full "app"lications (eg. monitoring components, CNI plugins, etc.)

I am committing a version with that change in the interest of time, but please feel free to correct again if you see it differently.